types/joint.d.ts

```diff
@@ -3041,6 +3041,66 @@ export namespace layout {
 
         export function fromGraphLib(glGraph: any, opt?: { [key: string]: any }): dia.Graph;
     }
+
+    export namespace Port {
+
+        interface PortOptions {
+            x?: number;
+            y?: number;
+            dx?: number;
+            dy?: number;
+            angle?: number;
+            start?: g.Point;
+            end?: g.Point;
+            startAngle?: number;
+            step?: number;
+            compensateRotation?: boolean;
+        }
+
+        type PortLayoutFunction = (ports: Array<Object>, elBBox: g.Rect, opt?: PortOptions) => Array<g.Point>;
```

Comment:
`opt` here is not optional. JointJS always pass the option object. Same for LabelLayout callback.